spanner/ddl/ast.go

```diff
@@ -215,49 +224,50 @@ func (pk IndexKey) PrintIndexKey(c Config) string {
 //	   [ CONSTRAINT constraint_name ]
 //		  FOREIGN KEY ( column_name [, ... ] ) REFERENCES ref_table ( ref_column [, ... ] ) }
 type Foreignkey struct {
-	Name         string
-	Columns      []string
-	ReferTable   string
-	ReferColumns []string
-	Id           string
+	Name           string
+	ColIds         []string
+	ReferTableId   string
+	ReferColumnIds []string
+	Id             string
 }
 
 // PrintForeignKey unparses the foreign keys.
 func (k Foreignkey) PrintForeignKey(c Config) string {
 	var cols, referCols []string
-	for i, col := range k.Columns {
+	for i, col := range k.ColIds {
 		cols = append(cols, c.quote(col))
-		referCols = append(referCols, c.quote(k.ReferColumns[i]))
+		referCols = append(referCols, c.quote(k.ReferColumnIds[i]))
 	}
 	var s string
 	if k.Name != "" {
 		s = fmt.Sprintf("CONSTRAINT %s ", c.quote(k.Name))
 	}
-	return s + fmt.Sprintf("FOREIGN KEY (%s) REFERENCES %s (%s)", strings.Join(cols, ", "), c.quote(k.ReferTable), strings.Join(referCols, ", "))
+	return s + fmt.Sprintf("FOREIGN KEY (%s) REFERENCES %s (%s)", strings.Join(cols, ", "), c.quote(k.ReferTableId), strings.Join(referCols, ", "))
 }
 
 // CreateTable encodes the following DDL definition:
 //
 //	create_table: CREATE TABLE table_name ([column_def, ...] ) primary_key [, cluster]
 type CreateTable struct {
-	Name     string
-	ColNames []string             // Provides names and order of columns
-	ColDefs  map[string]ColumnDef // Provides definition of columns (a map for simpler/faster lookup during type processing)
-	Pks      []IndexKey
-	Fks      []Foreignkey
-	Indexes  []CreateIndex
-	Parent   string //if not empty, this table will be interleaved
-	Comment  string
-	Id       string
+	Name        string
+	ColIds      []string             // Provides names and order of columns
+	ColDefs     map[string]ColumnDef // Provides definition of columns (a map for simpler/faster lookup during type processing)
+	PrimaryKeys []IndexKey
+	ForeignKeys []Foreignkey
+	Indexes     []CreateIndex
+	ParentId    string //if not empty, this table will be interleaved
+	Comment     string
+	Id          string
 }
 
 // PrintCreateTable unparses a CREATE TABLE statement.
-func (ct CreateTable) PrintCreateTable(config Config) string {
+func (ct CreateTable) PrintCreateTable(s Schema, config Config) string {
 	var col []string
 	var colComment []string
 	var keys []string
-	for _, cn := range ct.ColNames {
-		s, c := ct.ColDefs[cn].PrintColumnDef(config)
+	for _, colId := range ct.ColIds {
+
+		s, c := ct.ColDefs[colId].PrintColumnDef(config)
```

Comment:
we have used s Schema as method param and reassigning the value to it. where are we using s Schema?

Reply:
Actually they are different but there is confusion because of having the same name. Renamed the s param to spSchema and its used for extracting interleave parent table. Please review it.